db/compaction/compaction_iterator.cc

```diff
@@ -592,8 +592,11 @@ void CompactionIterator::NextFromInput() {
       // have hit (A)
       // We encapsulate the merge related state machine in a different
       // object to minimize change to the existing flow.
-      Status s = merge_helper_->MergeUntil(input_, range_del_agg_,
-                                           prev_snapshot, bottommost_level_);
+      auto allow_ingest_behind =
```

Comment:
@ajkr I'm trying to access `allow_ingest_behind` options from `compaction proxy` here in `compaction iterator` but it's not always present. Do you think this is a viable approach? Thanks!

Reply:
It looks reasonable as the proxy should be non-nullptr whenever the `CompactionIterator` is used for compaction. `CompactionIterator` is also used for flush, in which case proxy can be nullptr, but MERGE->PUT conversion doesn't happen there.

Although, the new code is a bit confusing when it has a variable `allow_ingest_behind == false` during a flush even though the DB may be configured with `allow_ingest_behind == true`.

How about folding this logic into the computation for `bottommost_level_`? It is reasonable to say the `CompactionIterator` is not targeting the bottommost level whenever the DB has `allow_ingest_behind` is set, since we have no way to guarantee something isn't ingested under the compaction output file later.